### PR TITLE
Add submission type property and filter

### PR DESCRIPTION
**Pull request by rrchai**
- https://github.com/Sage-Bionetworks/rocc-schemas/issues/205

Submission Types:
- DockerImage
- PredictionFile
- Other

### CODE REVIEW

File: openapi/components/parameters/ChallengeSubmissionType.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+description: The submission type of the challenge
+type: string
+enum:
+  - Prediction files
+  - Docker images
```
Comment (rrchai) [inline openapi/components/parameters/ChallengeSubmissionType.yaml:5]: Would you prefer "data-to-model vs model-to-data" or "Prediction files vs Docker images". I feel latter one is more user friendly?

Comment (tschaffter) [reply openapi/components/parameters/ChallengeSubmissionType.yaml:5]: I agree, the second extension is more descriptive. It is also more extensible. E.g. we could have "Singularity image" in the future. For the type value, I propose to adopt singular value.

Submission Type:

- Prediction file
- Docker image

File: openapi/components/schemas/ChallengeCreateRequest.yaml [hunk 2]
```diff
@@ -28,6 +28,11 @@ properties:
   platformId:
     $ref: ../parameters/ChallengePlatformId.yaml
     nullable: true
+  submissionTypes:
+    type: array
```
Comment (tschaffter) [inline openapi/components/schemas/ChallengeCreateRequest.yaml:32]: Good idea!

File: openapi/components/parameters/ChallengeSubmissionType.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+description: The submission type of the challenge
+type: string
+enum:
+  - Prediction files
```
Comment (tschaffter) [inline openapi/components/parameters/ChallengeSubmissionType.yaml:4]: I propose singular values

File: openapi/components/schemas/ChallengeCreateRequest.yaml [hunk 4]
```diff
@@ -50,7 +55,10 @@ example:
   startDate: "2020-11-10"
   endDate: "2020-12-31"
   platformId: 507f1f77bcf86cd799439011
+  submissionTypes:
+    - Prediction files
```
Comment (tschaffter) [inline openapi/components/schemas/ChallengeCreateRequest.yaml:59]: Use singular values

**Review by tschaffter — COMMENTED** [review]
Minor modifications.

File: openapi/components/parameters/ChallengeSubmissionType.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+description: The submission type of the challenge
+type: string
+enum:
+  - Docker image
+  - Prediction file
+  - Research paper
```
Comment (rrchai) [inline openapi/components/parameters/ChallengeSubmissionType.yaml:6]: @tschaffter,
[CAMI](http://camda2018.bioinf.jku.at/doku.php/call_for_papers) requires participants to submit the full method paper + optional abstract. If I understand correctly, they do not require prediction files or codes for submission. So I add "Research paper" (not sure if writeup is a better word) method as well here.

Comment (rrchai) [reply openapi/components/parameters/ChallengeSubmissionType.yaml]: I replace "Research paper" with "Other" now.

Comment (tschaffter) [reply openapi/components/parameters/ChallengeSubmissionType.yaml]: @rrchai Enum values should be snake or camel case. Either is fine for now (GitHub OpenAPI spec mixes both).

- docker_image
- prediction_file
- research_paper

Comment (rrchai) [reply openapi/components/parameters/ChallengeSubmissionType.yaml]: cool! Thanks for reminding. I have updated to use camel case.

notes for future use:
```
lodash.startCase(str);
// Example:
_.startCase('helloThere');
// ➜ 'Hello There'
```

Comment (tschaffter) [reply openapi/components/parameters/ChallengeSubmissionType.yaml]: Nice!

File: openapi/components/schemas/ChallengeCreateRequest.yaml (hunk 4 above)
Comment (tschaffter) [inline openapi/components/schemas/ChallengeCreateRequest.yaml:59]: Examples need to be updated

Comment (rrchai) [reply openapi/components/schemas/ChallengeCreateRequest.yaml]: good catch!

**Review by tschaffter — APPROVED** [review]
There is a merge conflict. Feel free to merge once fixed.

**Pull request merged**